packages/cli/src/actions/init.ts

```diff
@@ -0,0 +1,85 @@
+import path from 'path';
+import { execSync as exec } from 'child_process';
+import fse from 'fs-extra';
+import ora from 'ora';
+import { prompt } from 'inquirer';
+import { sync } from 'command-exists';
+import * as os from 'os';
+
+import { InitCommandHandler } from '../types';
+import { optionalNpmClients, daemonPackage, boardPackage } from '../config';
+
+const spinner = ora();
+const NPM_PUBLIC_REGISTRY = 'https://registry.npmjs.org/';
+
+function npmInstall(npmClient: string, name: string, beta: boolean, cwd: string, env: NodeJS.ProcessEnv) {
+  if (beta) {
+    name += '@beta';
+  }
+  const command = `${npmClient} install ${name} --registry ${NPM_PUBLIC_REGISTRY}`;
+  console.info(`exec ${command}`);
+  return exec(command, { cwd, env, stdio: 'inherit' });
+}
+
+/**
+ * install all dependencies of pipcook into working dir
+ */
+const init: InitCommandHandler = async ({ client, beta, tuna }) => {
+  let npmClient = 'npm';
+  const npmInstallEnvs = Object.assign({}, process.env);
+  if (tuna) {
+    npmInstallEnvs.BOA_CONDA_INDEX = 'https://pypi.tuna.tsinghua.edu.cn/simple';
+    npmInstallEnvs.BOA_CONDA_MIRROR = 'https://mirrors.tuna.tsinghua.edu.cn/anaconda/miniconda';
+  }
+  if (client) {
+    npmClient = client;
+    if (!optionalNpmClients.includes(npmClient)) {
+      spinner.fail(`invalid npm client: ${npmClient}.`);
+      return process.exit(1);
+    }
+  } else {
+    const clientChoices = [];
+    for (const npmClient of optionalNpmClients) {
+      if (sync(npmClient)) {
+        clientChoices.push(npmClient);
+      }
+    }
+
+    if (clientChoices.length === 1) {
+      npmClient = clientChoices[0];
+    } else if (clientChoices.length > 1) {
+      const answer = await prompt([
+        {
+          type: 'list',
+          name: 'client',
+          message: 'which client do you want to use?',
+          choices: clientChoices
+        }
+      ]);
+      npmClient = answer.client;
+    } else {
+      spinner.fail(`no npm client detected`);
+      return process.exit(1);
+    }
+  }
+
+  // Install the daemon and pipboard.
+  // use ~/.pipcook as PIPCOOK_DIR
+  const PIPCOOK_DIR = path.join(os.homedir(), '.pipcook');
+  const DAEMON_DIR = path.join(PIPCOOK_DIR, 'server');
+  const BOARD_DIR = path.join(PIPCOOK_DIR, 'pipboard');
+
+  try {
+    await fse.ensureDir(DAEMON_DIR);
+    await fse.ensureDir(BOARD_DIR);
+    npmInstall(npmClient, daemonPackage, beta, DAEMON_DIR, npmInstallEnvs);
```

Comment:
Remember we cannot use tnpm to install daemon because the midway version is not correct with tnpm. Therefore I made a check here that if user choose tnpm, i will replace it with npm

Reply:
Do we need to avoid tnpm?

Reply:
Not quite sure about this. As far as i know, lots of developers inside Alibaba like using tnpm which is much faster with company intranet.

Reply:
Ok, we should remove the tnpm on our community edition :)